### DISCUSSION TITLE
Functions list: mismatches of functions prototypes

### DISCUSSION THREAD

**Discussion by erdetn** [vlang/v#11318] [General]
Hello,

I was testing with functions handlers (list of functions). `math_ops := [add, div]`. The first case was with two function having the same function prototypes - `fn (f32, f32) f32`, and the other case was `fn (int, int) f32` for `add` function, and `fn (f32, f32) f32` for the `div` function.
**1st case**
```
module main
fn add(a f32, b f32) f32 {
    return (a + b)
}
fn div(a f32, b f32) f32 {
    println('div: $a $b')
    return (a / b)
}
fn main() {
    fns := [add, div]
    println(fns[0](10.0, 5.0))  // 15.0
    println(fns[1](10.0, 5.0))  //  2.0
}
```
and the **2nd case**
```
module main
fn add(a int, b int) f32 {
    return (a + b)
}
fn div(a f32, b f32) f32 {
    println('div: $a $b')
    return (a / b)
}
fn main() {
    fns := [add, div]
    println(fns[0](10.0, 5.0))  // 15.0
    println(fns[1](10.0, 5.0))  //  1.0
}
```
Assumption would be that the output from`div` is `2.0`, but the output is `1.0`.  In addition, by printing out corresponding arguments, I am not getting the given values: `div: 1.1e-44 1.1e-44`.
Is this because of functions prototypes, mismatches which is leading to this incorrectness?
Do we have any compile flag for further check to save the developer from this behavior?

**Comment by spytheman** [comment]
The second case should have been caught by V's checker stage, and it should have produced an error about the mismatched fn types.
Please file an issue.

**Reply by erdetn** [reply]
@spytheman thanks for checking. Submitted as an issue.
